libs/remixd/src/services/gitClient.ts

```diff
@@ -0,0 +1,62 @@
+import * as WS from 'ws' // eslint-disable-line
+import { PluginClient } from '@remixproject/plugin'
+const { spawn } = require('child_process')
+const gitRegex = '^git\\s[^&|;]*$'
+
+export class GitClient extends PluginClient {
+  methods: ['command']
+  websocket: WS
+  currentSharedFolder: string
+  readOnly: boolean
+
+  setWebSocket (websocket: WS): void {
+    this.websocket = websocket
+  }
+
+  sharedFolder (currentSharedFolder: string, readOnly: boolean): void {
+    this.currentSharedFolder = currentSharedFolder
+    this.readOnly = readOnly
+  }
+
+  command (cmd: string) {
+    return new Promise((resolve, reject) => {
+      try {
+        try {
```

Comment:
The double try / catch is redundant You can remove it as if validateCommand fails it will be catch in the first try and reject the promise